packages/discord.js/typings/index.d.ts

```diff
@@ -6107,22 +6107,28 @@ export interface PartialChannelData {
 
 export type Partialize<
   T extends AllowedPartial,
-  N extends keyof T | null = null,
-  M extends keyof T | null = null,
-  E extends keyof T | '' = '',
+  NulledKeys extends keyof T | null = null,
+  NullableKeys extends keyof T | null = null,
+  OverridableKeys extends keyof T | '' = '',
 > = {
-  readonly client: Client<true>;
-  id: Snowflake;
-  partial: true;
-} & {
-  [K in keyof Omit<T, 'client' | 'id' | 'partial' | E>]: K extends N ? null : K extends M ? T[K] | null : T[K];
+  [K in keyof Omit<T, OverridableKeys>]: K extends 'partial'
+    ? true
+    : K extends NulledKeys
+    ? null
+    : K extends NullableKeys
+    ? T[K] | null
+    : T[K];
 };
```

Comment:
When I said "another solution", I didn't mean to apply it to the existing solution. I just proposed an alternative way of doing it, so it's one or the other. If you apply the solution I proposed, all the changes below can be reverted as they are unnecessary.